canard.c

```diff
@@ -332,10 +331,15 @@ int16_t canardHandleRxFrame(CanardInstance* ins, const CanardCANFrame* frame, ui
         releaseStatePayload(ins, rx_state);
         if (!IS_START_OF_TRANSFER(tail_byte))
         {
-            rx_state->transfer_id++;
```

Comment:
There doesn't seem to be a reason to remove this now.

Reply:
The utility of the extra reporting aside I think the changes suggested result in some bugs, and they cause some of the tests to fail:

```
/home/mike/Projects/dsdl_canard_fix/branch/libcanard/tests/test_rxerr.cpp:402: FAILED:
  REQUIRE( err == -18 )
with expansion:
  0 == -18
```

This case is making sure that we can't receive two successive packets in quick succession with the same TID. It only occurs as an issue on single frame transfers:

```
    frame.data[0] = CONSTRUCT_TAIL_BYTE(1, 1, 1, 1);
    frame.id = 0 | CANARD_CAN_FRAME_EFF;            //Set EFF
    frame.id = CONSTRUCT_SVC_ID(0, 0, 1, 20, 0);
    frame.data_len = 1;
    err = canardHandleRxFrame(&canard, &frame, 1);
    REQUIRE(err == CANARD_OK);

    //More messages with same TID fail
    err = canardHandleRxFrame(&canard, &frame, 2);
    REQUIRE(err == -CANARD_ERROR_RX_DUPLICATE_TID);
```

If we leave that case in, but still remove the `transfer_id++` calls then we get some different failures - we detect duplicate packets when a well behaved caller is incrementing its TID :

```
    //1 Frame packet, should pass
    frame.data[0] = CONSTRUCT_TAIL_BYTE(1, 1, 0, 0);
    frame.id = CONSTRUCT_SVC_ID(0, 0, 1, 20, 0);
    frame.data_len = 1;
    err = canardHandleRxFrame(&canard, &frame, 1);
    REQUIRE(CANARD_OK == err);

    //Send a start packet, should pass
    frame.data[7] = CONSTRUCT_TAIL_BYTE(1, 0, 0, 1);

    frame.id = CONSTRUCT_SVC_ID(0, 0, 1, 20, 0);
    frame.data_len = 8;                             //Data length MUST be full packet
    err = canardHandleRxFrame(&canard, &frame, 1);
    REQUIRE(CANARD_OK == err);
```

```
-------------------------------------------------------------------------------
canardHandleRxFrame no state handling, Correctness
-------------------------------------------------------------------------------
/home/mike/Projects/dsdl_canard_fix/branch/libcanard/tests/test_rxerr.cpp:201
...............................................................................

/home/mike/Projects/dsdl_canard_fix/branch/libcanard/tests/test_rxerr.cpp:242: FAILED:
  REQUIRE( 0 == err )
with expansion:
  0 == -18
```

There might be something more subtle going on, or maybe the tests aren't following the spec right but I think the PR is correct. I can make the case report the `UNEXPECTED_TID` code but I think the case and no longer incrementing the `transfer_id` need to stay